Build failure: xdg-desktop-portal-hyprland-1.3.3

### Steps To Reproduce

Steps to reproduce the behavior:
1. build *xdg-desktop-portal-hyprland-1.3.3*

### Build log

```
Running phase: qtPreHook
@nix { "action": "setPhase", "phase": "qtPreHook" }
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/79n1c2k1n6cs6wzc1yw9j968z5f8k6ps-source
source root is source
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
fixing cmake files...
cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/lwrmvnk3iif67xm52a9xzvr8w0bkrjq8-xdg-desktop-portal-hyprland-1.3.3/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/lwrmvnk3iif67xm52a9xzvr8w0bkrjq8-xdg-desktop-portal-hyprland-1.3.3/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/lwrmvnk3iif67xm52a9xzvr8w0bkrjq8-xdg-desktop-portal-hyprland-1.3.3/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/lwrmvnk3iif67xm52a9xzvr8w0bkrjq8-xdg-desktop-portal-hyprland-1.3.3/share/doc/xdg-desktop-portal-hyprland -DCMAKE_INSTALL_INFODIR=/nix/store/lwrmvnk3iif67xm52a9xzvr8w0bkrjq8-xdg-desktop-portal-hyprland-1.3.3/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/lwrmvnk3iif67xm52a9xzvr8w0bkrjq8-xdg-desktop-portal-hyprland-1.3.3/share/man -DCMAKE_INSTALL_OLDINCLUDEDIR=/nix/store/lwrmvnk3iif67xm52a9xzvr8w0bkrjq8-xdg-desktop-portal-hyprland-1.3.3/include -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/lwrmvnk3iif67xm52a9xzvr8w0bkrjq8-xdg-desktop-portal-hyprland-1.3.3/include -DCMAKE_INSTALL_SBINDIR=/nix/store/lwrmvnk3iif67xm52a9xzvr8w0bkrjq8-xdg-desktop-portal-hyprland-1.3.3/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/lwrmvnk3iif67xm52a9xzvr8w0bkrjq8-xdg-desktop-portal-hyprland-1.3.3/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/lwrmvnk3iif67xm52a9xzvr8w0bkrjq8-xdg-desktop-portal-hyprland-1.3.3/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_OSX_SYSROOT= -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/gxpdysvlj9l6bfin03k6yq0s922pk7pg-gcc-wrapper-13.3.0/bin/strip -DCMAKE_RANLIB=/nix/store/gxpdysvlj9l6bfin03k6yq0s922pk7pg-gcc-wrapper-13.3.0/bin/ranlib -DCMAKE_AR=/nix/store/gxpdysvlj9l6bfin03k6yq0s922pk7pg-gcc-wrapper-13.3.0/bin/ar -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_INSTALL_PREFIX=/nix/store/lwrmvnk3iif67xm52a9xzvr8w0bkrjq8-xdg-desktop-portal-hyprland-1.3.3
-- The C compiler identification is GNU 13.3.0
-- The CXX compiler identification is GNU 13.3.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /nix/store/gxpdysvlj9l6bfin03k6yq0s922pk7pg-gcc-wrapper-13.3.0/bin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/gxpdysvlj9l6bfin03k6yq0s922pk7pg-gcc-wrapper-13.3.0/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring XDPH in Release with CMake
-- Checking deps...
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Performing Test HAVE_STDATOMIC
-- Performing Test HAVE_STDATOMIC - Success
-- Found WrapAtomic: TRUE
-- Found OpenGL: /nix/store/jz5imlpjk5a01224mimn6bsp9v0wrlgb-libglvnd-1.7.0/lib/libOpenGL.so
-- Found WrapOpenGL: TRUE
-- Found XKB: /nix/store/rnlflxkgglrdiii7wl7k99893zr4iamc-libxkbcommon-1.7.0/lib/libxkbcommon.so (found suitable version "1.7.0", minimum required is "0.5.0")
-- Found WrapVulkanHeaders: /nix/store/v0qw5vwbbhz0s32g8prvgpd8zy26vziz-vulkan-headers-1.3.283.0/include
-- Found PkgConfig: /nix/store/fcmc82gp5qc0s8qvcw4c0w46c67zd295-pkg-config-wrapper-0.29.2/bin/pkg-config (found version "0.29.2")
-- Checking for modules 'wayland-client;wayland-protocols;wayland-scanner;libpipewire-0.3;libspa-0.2;libdrm;gbm;hyprlang>=0.2.0'
--   Found wayland-client, version 1.23.0
--   Found wayland-protocols, version 1.36
--   Found wayland-scanner, version 1.23.0
--   Found libpipewire-0.3, version 1.2.3
--   Found libspa-0.2, version 0.2
--   Found libdrm, version 2.4.122
--   Found gbm, version 24.2.1
--   Found hyprlang, version 0.5.2
-- Checking for module 'sdbus-c++'
--   Found sdbus-c++, version 1.5.0
-- Checking for module 'hyprland-protocols'
--   Found hyprland-protocols, version 0.3.0
-- Found WaylandScanner at /nix/store/mq0445zflcxkh6r38afzczlpw8bax9kv-wayland-scanner-1.23.0-bin/bin/wayland-scanner
-- Found wayland-protocols at //nix/store/sm9mxbw5lanjmabjgd53yps0b9j3lrc9-wayland-protocols-1.36/share/wayland-protocols
-- Configuring done (1.9s)
[31mCMake Error in CMakeLists.txt:
  Imported target "PkgConfig::deps" includes non-existent path

    "/nix/store/0ikl83z4h69imxkwsly3hcm424f7wi7j-wayland-scanner-1.23.0-dev/include"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.


[0m
[31mCMake Error in CMakeLists.txt:
  Imported target "PkgConfig::deps" includes non-existent path

    "/nix/store/0ikl83z4h69imxkwsly3hcm424f7wi7j-wayland-scanner-1.23.0-dev/include"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.


[0m
-- Generating done (0.0s)
[33mCMake Warning:
  Manually-specified variables were not used by the project:

    BUILD_TESTING
    CMAKE_EXPORT_NO_PACKAGE_REGISTRY
    CMAKE_POLICY_DEFAULT_CMP0025

[0m
[0mCMake Generate step failed.  Build files cannot be regenerated correctly.[0m
```

### Additional context

This issue only recently occurred upon updating `nixpkgs-unstable` to the latest revision.
My entire nix config is [here](https://github.com/fdnt7/nix-config).

### Notify maintainers

@fufexan 

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.47, NixOS, 24.11 (Vicuna), 24.11.20240828.71e91c4`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.5`
 - nixpkgs: `/nix/store/887hpp8a2i99n9jjwcvz6qkhhhqsvzkg-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/applications/window-managers/hyprwm/xdg-desktop-portal-hyprland/default.nix
+++ b/pkgs/applications/window-managers/hyprwm/xdg-desktop-portal-hyprland/default.nix
@@ -36,6 +36,29 @@ stdenv.mkDerivation (finalAttrs: {
     hash = "sha256-cyyxu/oj4QEFp3CVx2WeXa9T4OAUyynuBJHGkBZSxJI=";
   };
 
+  patches = [
+    # CMake formatting, required for the next two commits to cleanly apply
+    (fetchpatch {
+      url = "https://github.com/hyprwm/xdg-desktop-portal-hyprland/commit/5555f467f68ce7cdf1060991c24263073b95e9da.patch";
+      hash = "sha256-yNkg7GCXDPJdsE7M6J98YylnRxQWpcM5N3olix7Oc1A=";
+    })
+    # removes wayland-scanner from deps, as it includes a pkg-config that
+    # defines that it has a non-existent include directory which trips up CMake
+    (fetchpatch {
+      url = "https://github.com/hyprwm/xdg-desktop-portal-hyprland/commit/0dd9af698b9386bcf25d3ea9f5017eca721831c1.patch";
+      hash = "sha256-Y6eWASHoMXVN2rYJ1rs0jy2qP81/qbHsZU+6b7XNBBg=";
+    })
+    # handle finding wayland-scanner more nicely
+    (fetchpatch {
+      url = "https://github.com/hyprwm/xdg-desktop-portal-hyprland/commit/2425e8f541525fa7409d9f26a8ffaf92a3767251.patch";
+      hash = "sha256-6dCg/U/SIjtvo07Z3tn0Hn8Xwx72nwVz6Q2cFnObonU=";
+    })
+  ];
+
+  depsBuildBuild = [
+    pkg-config
+  ];
+
   nativeBuildInputs = [
     cmake
     pkg-config

